src/ape/api/networks.py

```diff
@@ -204,21 +204,75 @@ def set_default_network(self, network_name: str):
     def encode_deployment(
         self, deployment_bytecode: HexBytes, abi: ConstructorABI, *args, **kwargs
     ) -> "TransactionAPI":
-        ...
+        """
+        Create a deployment transaction in the given ecosystem.
+        This may require connecting to other networks.
+
+        Args:
+            deployment_bytecode (HexBytes): The bytecode to deploy.
+            abi (ConstructorABI): The constructor interface of the contract.
+            *args: Constructor arguments.
+            **kwargs: Transaction arguments.
+
+        Returns:
+            class:`~ape.api.transactions.TransactionAPI`
+        """
 
     @abstractmethod
     def encode_transaction(
         self, address: AddressType, abi: MethodABI, *args, **kwargs
     ) -> "TransactionAPI":
-        ...
+        """
+        Encode a transaction object from a contract function's abi and call arguments.
+        Update the transaction arguments with the overrides in ``kwargs`` as well.
+
+        Args:
+            address (AddressType): The address of the contract.
+            abi (MethodABI): The function to call on the contract.
+            *args: Function arguments.
+            **kwargs: Transaction arguments.
+
+        Returns:
+            class:`~ape.api.transactions.TransactionAPI`
+        """
 
     @abstractmethod
     def decode_logs(self, abi: EventABI, raw_logs: List[Dict]) -> Iterator[ContractLog]:
-        ...
+        """
+        Decode any contract logs that match the given event ABI from the raw log data.
+
+        Args:
+            abi (EventABI): The event producing the logs.
+            raw_logs (List[Dict]): A list of raw log data from the chain.
+
+        Returns:
+            Iterator[:class:`~ape.types.ContractLog`]
+        """
 
     @abstractmethod
     def create_transaction(self, **kwargs) -> "TransactionAPI":
-        ...
+        """
+        Create a transaction using key-value arguments.
+
+        Args:
+            **kwargs: Everything the transaction needs initialize.
+
+        Returns:
+            class:`~ape.api.transactions.TransactionAPI`
+        """
+
+    @abstractmethod
+    def decode_calldata(self, abi: MethodABI, raw_data: bytes) -> Tuple[Any, ...]:
```

Comment:
Should it just return `Any`? I think enforcing the tuple return might be too strict

Reply:
💯